web-frontend/src/pages/Assets/DeleteAssetModal.tsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+
+import {
+  Button,
+  Input,
+  PopupModal,
+  PopupModalContent,
+  PopupModalHeader,
+  toast,
+} from '@massalabs/react-ui-kit';
+import { AxiosError } from 'axios';
+import { FiTrash2 } from 'react-icons/fi';
+import { useParams } from 'react-router-dom';
+
+import {
+  DeleteAssetsErrors,
+  assetDeleteErrors,
+  deleteConfirm,
+} from '@/const/assets/assets';
+import { useDelete } from '@/custom/api';
+import Intl from '@/i18n/i18n';
+import { IToken } from '@/models/AccountModel';
+
+export function DeleteAssetModal({ ...props }) {
+  const { tokenAddress, setModalOpen, refetch } = props;
+  const { nickname } = useParams();
+
+  const [deletePhrase, setDeletePhrase] = useState<string>('');
+  const [error, setError] = useState<DeleteAssetsErrors | null>(null);
+
+  const {
+    mutate: mutateDelete,
+    isSuccess: isSuccessDelete,
+    isError: isErrorDelete,
+    error: errorDelete,
+  } = useDelete<IToken[]>(
+    `accounts/${nickname}/assets?assetAddress=${tokenAddress}`,
+  );
+
+  const axiosError = errorDelete as AxiosError;
+  const deleteErrorStatus = axiosError?.response?.status;
+
+  useEffect(() => {
+    if (isSuccessDelete) {
+      toast.success(Intl.t('assets.delete.success'));
+      setModalOpen(false);
+      refetch();
+    } else if (isErrorDelete) {
+      displayErrors(deleteErrorStatus);
+    }
+  }, [isSuccessDelete, isErrorDelete]);
+
+  function displayErrors(postStatus: number | undefined) {
+    switch (postStatus) {
+      case assetDeleteErrors.badRequest:
+        toast.error(Intl.t('assets.delete.bad-request'));
+        break;
+      case assetDeleteErrors.invalidAddress:
+        toast.error(Intl.t('assets.delete.invalid-address'));
+        break;
+      case assetDeleteErrors.serverError:
+        toast.error(Intl.t('assets.internal-server-error'));
+        break;
+      default:
+        toast.error(Intl.t('assets.unkown-error'));
+        console.log('Unknown Error:', deleteErrorStatus);
+    }
+  }
+
+  function validate(phrase: string) {
+    if (!phrase) {
+      setError({ phrase: Intl.t('assets.delete.no-value') });
+      return false;
+    }
+
+    if (phrase !== deleteConfirm) {
+      setError({ phrase: Intl.t('assets.delete.wrong-value') });
+      return false;
+    }
+    return true;
+  }
+
+  function confirmDelete() {
+    if (!validate(deletePhrase)) return;
+
+    mutateDelete({} as IToken[]);
```

Comment:
you can also simplify this, try:
```suggestion
    mutateDelete();
```

Reply:
<img width="581" alt="Screenshot 2023-08-10 at 12 12 48" src="https://github.com/massalabs/station-massa-wallet/assets/90157528/fc174a08-1d73-4a40-9e51-14dcb184ee86">